private/pkg/git/remote.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/bufbuild/buf/private/pkg/app"
+	"github.com/bufbuild/buf/private/pkg/command"
+)
+
+const (
+	bitBucketHostname           = "bitbucket"
+	githubHostname              = "github"
+	gitlabHostname              = "gitlab"
+	gitSuffix                   = ".git"
+	githubGitlabRemoteURLFormat = "https://%s%s/commit/%s"
```

Comment:
is it really worth it to smash these two cases together rather than having a separate variable for each, even if the value is the same?